backend/overstroomik_service/validators.py

```diff
@@ -0,0 +1,67 @@
+import logging
+
+from pydantic.class_validators import Validator
+
+from .auto_models import Data, FloodType, ProbabilityOfFlooding
+
+
+def validate_maximum_water_depth(cls, value):
+    """Return 0.0 for waterdepth in case of invalid values.
+
+    Invalid values are:
+    - everything lower than 0.01m (including -9999, -inf)
+    - everything above 50m as it's unrealistic (including inf)
+    """
+    # in the case that the value is a "100.0" we won't catch it
+    if isinstance(value, float):
+        if value < 0.01:
+            value = 0.0
+
+        elif value > 50.0:
+            value = 0.0
+
+    return value
+
+
+def validator_probability_of_flooding(cls, value):
+
+    # No action as the enum parse is passed
+    if isinstance(value, ProbabilityOfFlooding):
+        return value
+    else:
+        try:
+            return ProbabilityOfFlooding(value)
```

Comment:
@RolfWaterman You can actually just pass a string to an Enum. If that doesn't match, we need to change either the Data or the Enum.